From 96aef5ed96e02afe29db7e852bee7e22611a8839 Mon Sep 17 00:00:00 2001
From: Will Deacon <willdeacon@google.com>
Date: Wed, 22 Apr 2026 12:35:50 +0100
Subject: [PATCH 114/223] ANDROID: KVM: arm64: Fix rb_cpu_fits_desc()
 invocation from pKVM tracing

When initialising pKVM trace buffers, rb_cpu_fits_desc() is invoked
with the untrusted host virtual address for the trace descriptors
rather than the hypervisor virtual address. This makes subsequent bounds
comparisons nonsensical, so pass the hypervisor VA instead.

Cc: Vincent Donnefort <vdonnefort@google.com>
Signed-off-by: Will Deacon <willdeacon@google.com>
Bug: 503395542
Change-Id: Ic15a5d415a6510d45a6f04e9ca8174a6d0f58834
(cherry picked from commit dbced6416535141b9877882bd4775ca39ce94f79)
Signed-off-by: Lee Jones <joneslee@google.com>
---
 arch/arm64/kvm/hyp/nvhe/trace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/kvm/hyp/nvhe/trace.c b/arch/arm64/kvm/hyp/nvhe/trace.c
index 8f8bd28a12e9..30dfc34977ff 100644
--- a/arch/arm64/kvm/hyp/nvhe/trace.c
+++ b/arch/arm64/kvm/hyp/nvhe/trace.c
@@ -583,7 +583,7 @@ int __pkvm_load_tracing(unsigned long desc_hva, size_t desc_size)
 		unsigned int cpu;
 
 		ret = -EINVAL;
-		if (!rb_cpu_fits_desc(pdesc, desc_hva + desc_size))
+		if (!rb_cpu_fits_desc(pdesc, (unsigned long)desc + desc_size))
 			break;
 
 		cpu = pdesc->cpu;
-- 
2.50.1

